### PR TITLE
[EPMEDU-800]: Add edge-to-edge support

### PR DESCRIPTION
**Pull request by Fileek**
- Added TransparentSystemUi composable
- Rearranged system bar paddings to fill its background
- Extracted almost all Surfaces into one Surface in AnimealTheme
- Removed redundant showBackground argument in previews
- Fixed AnimalTab comment

| With navigation bar  | Without navigation bar  |
| ------------- | ------------- |
|  <video src="https://user-images.githubusercontent.com/83027107/185453537-c18afd97-9c20-4d1b-93e1-93ff73aba75b.mp4"> | <video src="https://user-images.githubusercontent.com/83027107/185453615-1d5b6ac8-74c7-4e0f-adf3-c97924cf9d53.mp4"> |

### CODE REVIEW

**Comment by Fileek** [comment]
https://user-images.githubusercontent.com/83027107/185403040-2c06cd2b-fb80-4aaf-88a5-b158dbbfec5c.mp4

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt [hunk 1]
```diff
@@ -34,12 +36,20 @@ fun AnimealTheme(
...
     content: @Composable () -> Unit
 ) {
+    val systemUiController = rememberSystemUiController()
     val colors = if (darkTheme) {
         DarkColorPalette
     } else {
         LightColorPalette
     }
 
+    LaunchedEffect(Unit) {
```
Comment (egorikftp) [inline library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt]: This should be inside MaterialTheme

Comment (egorikftp) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt]: Please check in dark theme

File: feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/HomeScreen.kt [hunk 2]
```diff
@@ -1,11 +1,7 @@
 package com.epmedu.animeal.tabs.presentation
 
 import android.Manifest
-import androidx.compose.foundation.layout.Arrangement
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.*
```
Comment (VEZE) [inline feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/HomeScreen.kt]: remove wildcard

File: feature/login/src/main/java/com/epmedu/animeal/login/phone/presentation/EnterPhoneScreenUI.kt [hunk 3]
```diff
@@ -1,10 +1,7 @@
 package com.epmedu.animeal.login.phone.presentation
 
 import android.content.res.Configuration.UI_MODE_NIGHT_YES
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.imePadding
-import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.*
```
Comment (VEZE) [inline feature/login/src/main/java/com/epmedu/animeal/login/phone/presentation/EnterPhoneScreenUI.kt]: remove wildcard

**Review by VEZE — CHANGES_REQUESTED** [review]

File: feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/HomeScreen.kt (hunk 2 above)
Comment (Fileek) [reply feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/HomeScreen.kt]: Done

File: feature/login/src/main/java/com/epmedu/animeal/login/phone/presentation/EnterPhoneScreenUI.kt (hunk 3 above)
Comment (Fileek) [reply feature/login/src/main/java/com/epmedu/animeal/login/phone/presentation/EnterPhoneScreenUI.kt]: Done

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt (hunk 1 above)
Comment (Fileek) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt]: > This should be inside MaterialTheme

Done

Comment (Fileek) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt]: > Please check in dark theme

Recorded videos with navigation bar and without it in both themes

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt [hunk 4]
```diff
@@ -43,7 +47,20 @@ fun AnimealTheme(
     MaterialTheme(
         colors = colors,
         typography = Typography,
-        shapes = Shapes,
-        content = content,
-    )
+        shapes = Shapes
+    ) {
+        LaunchedEffect(Unit) {
```
Comment (egorikftp) [inline library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt]: Maybe better to split setup systemUiController out of Theme

Comment (Fileek) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt]: Where? In MainActivity?

Comment (egorikftp) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt]: 1. Leave `Surface` inside theme
2. Remove `Surface` from @Preview
3. Extact `systemUiController` into separate `Composable`

Comment (Fileek) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/theme/Theme.kt]: Done

**Review by VEZE — APPROVED** [review]

**Review by n3gbx — APPROVED** [review]

**Review by egorikftp — APPROVED** [review]

**Pull request merged**